### DISCUSSION TITLE
is it possible to create a rust type to add diesel derives as superset for the source type

### DISCUSSION THREAD

**Discussion by sergeken** [diesel-rs/diesel#3362] [Q&A] [answered]
I was wondering if there is a way to do something like this?
```
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Client {
	pub id: i64,
	pub first_name: String,
	pub last_name: String,
	pub address_line_1: String,
	pub address_line_2: String,
	timestamp: NaiveDateTime,
}
```
```
#[derive(Debug, Clone, Serialize, Deserialize, Queryable, Identifiable)]
#[diesel(table_name = client)]
struct DBClient(Client);
```
The Identifiable derives fails.

What I want to achieve is to have the "application types" free of diesel annotations.

**Comment by weiznich** [comment, marked as answer]
That's not possible with using the derives as these need access to the underlying fields. They do not have access to the necessary information on the new type wrapper.

You can write the corresponding implementations manually if you really want to use something like this. (I wouldn't recommend that)

**Reply by sergeken** [reply]
Thank you,
I was afraid for that answer.
Could there be a way to create an additional annotation to hint that the fields are in "sub type"

**Reply by weiznich** [reply]
Just an additional annotation that fields are in the "sub type" will not help here. Proc macros cannot access structural information about the sub type in any way. The only information they get there is that there is one inner field of the type `Client`, not how to construct client or which fields are necessary for that. Technically you could  add all that information to the outer type via annotations, but that would basically duplicate the definition of `Client` (or at least closely mirror a manual implementation of the corresponding trait.)

**Reply by sergeken** [reply]
Make sense
